test/test_utils_log.cpp

```diff
@@ -0,0 +1,352 @@
+// Copyright (C) 2024 Intel Corporation
+// Under the Apache License v2.0 with LLVM Exceptions. See LICENSE.TXT.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include "base.hpp"
+#include "test_helpers.h"
+#include <thread>
+
+#define MOCK_FILE_PTR (FILE *)0xBADBEEF
+std::string expected_filename;
+int expect_fopen_count = 0;
+int fopen_count = 0;
+FILE *mock_fopen(const char *filename, const char *mode) {
+    fopen_count++;
+    EXPECT_STREQ(filename, expected_filename.c_str());
+    EXPECT_STREQ(mode, "w");
+    return MOCK_FILE_PTR;
+}
+
+std::string expected_message = "";
+FILE *expected_stream;
+int expect_fput_count = 0;
+int fput_count = 0;
+int mock_fputs(const char *s, FILE *stream) {
+    fput_count++;
+    if (!expected_message.empty()) {
+        EXPECT_STREQ(s, expected_message.c_str());
+    }
+    EXPECT_EQ(stream, expected_stream);
+    return strlen(s);
+}
+
+int expect_fflush_count = 0;
+int fflush_count = 0;
+int mock_fflush(FILE *stream) {
+    fflush_count++;
+    EXPECT_EQ(stream, expected_stream);
+    return 0;
+}
+extern "C" {
+#define fopen(A, B) mock_fopen(A, B)
+#define fputs(A, B) mock_fputs(A, B)
+#define fflush(A) mock_fflush(A)
+
+const char *env_variable;
+//getenv returs 'char *' not 'const char *' so we need explicit cast to drop const
+#define getenv(X) strstr(X, "UMF_LOG") ? (char *)env_variable : getenv(X)
+#include "utils/utils_log.c"
+#undef getenv
+#undef fopen
+#undef fputs
+#undef fflush
+}
+using umf_test::test;
+
+TEST_F(test, matchEnvVar) {
```

Comment:
Could you add some tests for empty string and string with `;` at the beginning?

Reply:
Done